src/mkfs.fat.c

```diff
@@ -1795,5 +1795,13 @@ int main(int argc, char **argv)
 
     write_tables();		/* Write the filesystem tables away! */
 
+    if (fsync(dev) < 0) {
+        /* Let's make sure to sync the block device. Otherwise, if we operate on a loop device and people issue
+         * "losetup -d" right after this command finishes our in-flight writes might never hit the disk */
+
+        pdie("unable to synchronize %s", device_name);
+        exit(1);
```

Comment:
`pdie` does not return, it calls `exit`, so do not call `exit` function in your patch second time. That's why I suggested you to use this `pdie` function to reduce copying code; look at `pdie` implementation.